src/data/write_init_files.py

```diff
@@ -966,11 +966,17 @@ def write_phys_read_subroutine(outfile, host_dict, host_vars, host_imports,
     outfile.write("const_props => cam_model_const_properties()", 7)
     outfile.write("constituent_has_default = .false.", 7)
     outfile.write("call const_props(constituent_idx)%has_default(constituent_has_default, constituent_errflg, constituent_errmsg)", 7)
+    outfile.write("if (constituent_errflg .ne. 0) then", 7)
+    outfile.write("call endrun(constituent_errmsg)", 8)
```

Comment:
I might use the symbolic representation of "not equal" here (as I believe it better matches the code style elsewhere), and I would include the file and line number along with the error message as well:

```suggestion
    outfile.write("if (constituent_errflg /= 0) then", 7)
    outfile.write("call endrun(constituent_errmsg, file=__FILE__, line=__LINE__)", 8)
```